include/cuco/detail/static_map.inl

```diff
@@ -91,14 +102,17 @@ void static_map<Key, Value, Scope, Allocator>::find(
   auto view             = get_device_view();
 
   detail::find<block_size, tile_size, Value>
-    <<<grid_size, block_size>>>(first, last, output_begin, view, hash, key_equal);
-  CUCO_CUDA_TRY(cudaDeviceSynchronize());
+    <<<grid_size, block_size, 0, stream>>>(first, last, output_begin, view, hash, key_equal);
+  // stream'd execution assumes sync not required
+  if (stream == 0)
+    CUCO_CUDA_TRY(cudaStreamSynchronize(stream)); // ensures legacy behavior
```

Comment:
We don't need this.

Reply:
I don't want to retain this same behavior because then the function is sometimes synchronous and sometimes asynchronous. If it takes a stream, it should be always assumed to be asynchronous.